From 62aba138a72c1201f25a2b0020694aa9f349033b Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 9 May 2019 16:14:17 +0800
Subject: [PATCH 26/30] drm/bridge: adv7511: fix adv7535 DSI-HDMI converter
 malfunction issue

1. When picking up the patch from SDK, there is typo (adv7511->addr_edid)
which cause compile error.
2. The function "adv7511_init_cec_regmap" should be called after the register
ADV7511_REG_CEC_I2C_ADDR is initialised with proper value.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpu/drm/bridge/adv7511/adv7511_drv.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
index a106883ab948..7269b36f9c56 100644
--- a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
+++ b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
@@ -1010,9 +1010,9 @@ static int adv7511_init_cec_regmap(struct adv7511 *adv)
 {
 	int ret;
 
-	if (adv7511->addr_edid != 0)
+	if (adv->addr_cec != 0)
 		adv->i2c_cec = i2c_new_secondary_device(adv->i2c_main, "cec",
-							adv7511->addr_edid);
+							adv->addr_cec);
 	else
 		adv->i2c_cec = i2c_new_secondary_device(adv->i2c_main, "cec",
 							ADV7511_CEC_I2C_ADDR_DEFAULT);
@@ -1231,10 +1231,6 @@ static int adv7511_probe(struct i2c_client *i2c, const struct i2c_device_id *id)
 		regmap_write(adv7511->regmap, ADV7511_REG_PACKET_I2C_ADDR,
 			     adv7511->i2c_packet->addr << 1);
 
-	ret = adv7511_init_cec_regmap(adv7511);
-	if (ret)
-		goto err_i2c_unregister_packet;
-
 	if (adv7511->addr_cec != 0)
 		regmap_write(adv7511->regmap, ADV7511_REG_CEC_I2C_ADDR,
 			     adv7511->addr_cec << 1);
@@ -1242,6 +1238,11 @@ static int adv7511_probe(struct i2c_client *i2c, const struct i2c_device_id *id)
 		regmap_write(adv7511->regmap, ADV7511_REG_CEC_I2C_ADDR,
 			     adv7511->i2c_cec->addr << 1);
 
+	ret = adv7511_init_cec_regmap(adv7511);
+	if (ret)
+		goto err_i2c_unregister_packet;
+
+
 	INIT_WORK(&adv7511->hpd_work, adv7511_hpd_work);
 
 	if (i2c->irq) {
-- 
2.17.1

